### PR TITLE
#340: Sizes update

### PR DESCRIPTION
**Pull request by AnyaVeselova**
# Description

**Relates #340 **  
This pull request:
- isolates sizes array initialization and sizes selection logic into a separate component for cleaner code, better separation of concerns, and easier updates or adjustments later
- adds more sizes option (including half shoe sizes)
- creates a new euShoeSizes array
- imports the component inside the form of a parent component

### Files changed

 - SizeSelector.tsx has been created
 - AddNewItemForm.tsx

### CODE REVIEW

File: components/messaging/ConversationCard.tsx
Comment (ShaughnAnderson94) [inline components/messaging/ConversationCard.tsx:1]: the shoe gird is looking good however this code relating to conversation card probably shouldn't be in this issue (be careful of the stuff we talked about earlier and just do a git status every now and then to see what you are committing/check your new branch is new)

**Review by ShaughnAnderson94 — CHANGES_REQUESTED** [review]

File: components/form/SizeSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    'UK12',
+    'UK13',
+    'UK14',
+    'UK15',
+    'UK16',
+  ];
+
+  const euShoeSizes = ['EU36', 'EU37', 'EU38', 'EU39', 'EU40', 'EU41', 'EU42'];
+
+  let sizeOptions: string[] = [];
```
Comment (nichgalzin) [inline components/form/SizeSelector.tsx:58]: I think the refactor of this logic to it's own component is great I really like the cleaning up of the AddNewItemForm component to be doing less. However the use of ```let``` here, while totally functional and probably fine for a simple case like this is probably better put into a local state variable using the react useState hook.

File: components/form/SizeSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+  if (category === 'clothing') {
```
Comment (nichgalzin) [inline components/form/SizeSelector.tsx:60]: Keeping in line then with the above logic you would then wrap this logic in the useEffect hook and put the category prop in the dependency array. Again what you did here is totally fine from a logic point of view, but this implementation will fit a bit better the React conventions we're using in the rest of the project.

**Review by nichgalzin — CHANGES_REQUESTED** [review]
Thanks a lot for this Anya! Left a few requests for changes. Let me know if you have any questions!

**Review by nichgalzin — CHANGES_REQUESTED** [review]
Hi Anna,
   This is all looking great. Can you please just remove the middleware.ts file from the this pr and then I can approve it? This looks just like an auto formatting thing and the code has been repeated. In any case this is unrelated to this PR and we want to avoid unintentional code changes to unrelated file. Thanks for this, fix this and we can merge this PR!

**Pull request closed without merging**